x-pack/plugins/triggers_actions_ui/public/application/components/builtin_alert_types/threshold/visualization.tsx

```diff
@@ -175,12 +175,10 @@ export const ThresholdVisualization: React.FunctionComponent<Props> = ({
     threshold,
     startVisualizationAt,
   ]);
-  /* eslint-enable react-hooks/exhaustive-deps */
 
   if (!charts || !uiSettings || !dataFieldsFormats) {
     return null;
   }
-  const chartsTheme = charts.theme.useChartsTheme();
 
```

Comment:
```suggestion
  const chartsTheme = charts.theme.useChartsTheme();
  const chartsBaseTheme = charts.theme.useChartsBaseTheme();
```

Reply:
To avoid the error that we currently get in the browser console:
```
Uncaught Error: Rendered more hooks than during the previous render.
    at updateWorkInProgressHook (react-dom.development.js:16411)
    at updateReducer (react-dom.development.js:16471)
    at updateState (react-dom.development.js:16640)
    at Object.useState (react-dom.development.js:17363)
    at useState (react.development.js:1619)
    at ThemeService.useChartsTheme (theme.ts:43)
    at ThresholdVisualization (visualization.tsx:218)
    at renderWithHooks (react-dom.development.js:16260)
    at updateFunctionComponent (react-dom.development.js:18347)
    at beginWork$1 (react-dom.development.js:20176)
```
we should only call hooks from the top level of a component, and never put them in conditional statements.

Reply:
Yup thanks I totally messed that up, sorry about that. This is what I get for trying to simplify hooks.

Reply:
That interesting, since `charts` may be undefined you could seemingly `'Rendered more hooks than during the previous render'` with the existing logic. Did the existing logic every throw this error?

```ts
if (!charts || !uiSettings || !dataFieldsFormats) {
  return null;
}
const chartsTheme = charts.theme.useChartsTheme(); // <-- may be undefined on first render
```

Reply:
Thanks for the review @YulNaumenko! I fixed all the things I broke here 6207f94. Let me know if there is anything else.

Reply:
Probably it makes sense to check if a `charts.theme` is undefined as well :), it's our luckiness we didn't meet this error before.

Reply:
It looks like if `charts` is defined `charts.theme` will always be defined.

Thinking about it, I bet `charts`, `uiSettings` and `dataFieldsFormats ` are always defined, even though they could be undefined. Because undoubtedly you should have seen this error with any hook beyond an if statement that caught and returned null in a previous render. Or at least that's my experience.

Who knows, but I'm glad it works 😄